monitoring/check_collectd_mlab_test.py

```diff
@@ -0,0 +1,379 @@
+    self.fake_collectd = FakeCollectd(self.fake_socket)
+    self.fake_collectd.start()
+
+  def tearDown(self):
+    self.assertTrue(self.fake_collectd.shutdown())
+
+  def testunit_sock_sendcmd_RETURNS_successfully(self):
+    sock = check_collectd_mlab.sock_connect(self.fake_socket)
+
+    returned_value = check_collectd_mlab.sock_sendcmd(sock, 'GETVAL "whatever"')
+
+    self.assertEqual(returned_value, 1)
+
+  def testunit_sock_sendcmd_WHEN_receive_bad_reply_RETURNS_zero(self):
+    self.fake_collectd.set_response_behavior('send_bad_reply')
+    sock = check_collectd_mlab.sock_connect(self.fake_socket)
+
+    returned_value = check_collectd_mlab.sock_sendcmd(sock, 'GETVAL "whatever"')
+
+    self.assertEqual(returned_value, 0)
+
+  def testunit_assert_collectd_responds_WHEN_sock_sendcmd_fails(self):
+    self.fake_collectd.set_response_behavior('send_empty_reply')
+    check_collectd_mlab.COLLECTD_PID = (
+        os.path.join(self._testdata_dir, 'fake_pid'))
+    check_collectd_mlab.COLLECTD_UNIXSOCK = self.fake_socket
+
+    self.assertRaises(
+        check_collectd_mlab.SocketSendCommandCriticalError,
+        check_collectd_mlab.assert_collectd_responds)
+
+  def testunit_assert_collectd_responds_WHEN_sock_readline_fails(self):
+    self.fake_collectd.set_response_behavior('close_after_reply')
+    check_collectd_mlab.COLLECTD_PID = (
+        os.path.join(self._testdata_dir, 'fake_pid'))
+    check_collectd_mlab.COLLECTD_UNIXSOCK = self.fake_socket
+
+    self.assertRaises(
+        check_collectd_mlab.SocketReadlineCriticalError,
+        check_collectd_mlab.assert_collectd_responds)
+
+
+class MlabNagiosTests(unittest.TestCase):
+
+  def setUp(self):
+    self._testdata_dir = os.path.join(
+        os.path.dirname(check_collectd_mlab.__file__), 'testdata')
+
+  def testunit_sock_connect_WHEN_no_socket_RAISES_CriticalError(self):
+    self.assertRaises(
+        check_collectd_mlab.SocketConnectionCriticalError,
+        check_collectd_mlab.sock_connect, 'no_socket_name')
+
+  def testunit_sock_readline_WHEN_socket_error_RETURNS_empty_string(self):
+    mock_sock = mock.Mock(spec_set=socket.socket)
+    mock_sock.recv.side_effect = socket.error('fake error')
+
+    self.assertRaises(
+        check_collectd_mlab.SocketReadlineCriticalError,
+        check_collectd_mlab.sock_readline, mock_sock)
+
+  def testunit_assert_collectd_installed_WHEN_bin_missing_RAISES_CriticalError(
+      self):
+    check_collectd_mlab.COLLECTD_BIN = 'does_not_exist'
+
+    self.assertRaises(
+        check_collectd_mlab.MissingBinaryCriticalError,
+        check_collectd_mlab.assert_collectd_installed)
+
+  def testunit_assert_collectd_installed_WHEN_nagios_bin_missing_CriticalError(
+      self):
+    check_collectd_mlab.COLLECTD_BIN = (
+        os.path.join(self._testdata_dir, 'fake_bin'))
+    check_collectd_mlab.COLLECTD_NAGIOS = (
+        os.path.join(self._testdata_dir, 'does_not_exist'))
```

Comment:
I think this is a good reason to put everything in `setUp`. It's not super obvious when reading it that fake_bin is supposed to exist and does_not_exist doesn't. Reading through it just now, I got confused because I thought since it's a test for missing files, neither file existed. If `setUp` included:

```
check_collectd_mlab.COLLECTD_BIN = os.path.join(self._testdata_dir, 'fake_bin')
check_collectd_mlab.COLLECTD_NAGIOS = os.path.join(self._testdata_dir, 'fake_bin')
check_collectd_mlab.COLLECTD_UNIXSOCK = 'fake_sock'
check_collectd_mlab.VSYSPATH_BACKEND = os.path.join(self._testdata_dir, 'fake_backend')
check_collectd_mlab.VSYSPATH_SLICE = os.path.join(self._testdata_dir, 'fake_slice')
 check_collectd_mlab.VSYSPATH_ACL = os.path.join(self._testdata_dir, 'fake_acl')
```

Then when you set things to `does_not_exist` it's obvious that they're the only component that doesn't exist.